src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1882,7 +1882,7 @@ final case class TStruct(fields: IndexedSeq[Field], override val required: Boole
         } ]
            |  Hint: use `drop' or `select' to remove these fields from one side""".stripMargin)
 
-    val newStruct = TStruct(fields ++ other.fields.map(f => f.copy(index = f.index + size)))
+    val newStruct = TStruct(fields ++ other.fields.map(f => f.copy(index = f.index + size)), required || other.required)
```

Comment:
I believe this modification of the `apply` method will compile:

```scala

  def apply(args: (String, Type)*): TStruct =
    apply(false, args:_*)

  def apply(required: Boolean, args: (String, Type)*): TStruct =
    ...
```

Reply:
done